### PR TITLE
sync : Apply filter jetpack_sync_comment_meta_whitelist

### LINKED ISSUE #8170: Add a filter for comment_meta_whitelist
There is a filter for `post_meta_whitelist`:

https://github.com/Automattic/jetpack/blob/5e4413cc3b690d3b5afacd714d39ec85539959b4/sync/class.jetpack-sync-defaults.php#L318-L329

Can we get something like this for `comment_meta_whitelist` as well?

https://github.com/Automattic/jetpack/blob/5e4413cc3b690d3b5afacd714d39ec85539959b4/sync/class.jetpack-sync-defaults.php#L331-L336

Without it, you have to hook into something like `option_jetpack_sync_settings_comment_meta_whitelist`.

### PR DESCRIPTION
**Pull request by Umangvaghela**
Fixes #8170

### CODE REVIEW

**Comment by Umangvaghela** [comment]
@gravityrail 

Sir , can you review this PR when  you get a chance and provide your feedback if I miss something.

Thanks

**Comment by gravityrail** [comment]
This is necessary but not sufficient to fix the issue, since you need to replace a bunch of places where we actually directly grab the comment meta whitelist from the DB and/or Sync Defaults file.

**Comment by Umangvaghela** [comment]
@gravityrail 

Thanks for review.

I found one place where we call filter instead of call variable. I follow the post meta list functionality which is look like same as comment meta list.

Thanks

File: sync/class.jetpack-sync-defaults.php [hunk 1]
```diff
@@ -335,6 +335,17 @@ public static function get_post_meta_whitelist() {
 		'hc_foreign_user_id'
 	);
 
+	public static function get_comment_meta_whitelist() {
+		/**
+		 * Filter the list of comment meta data that are manageable via the JSON API.
+		 *
+		 * @module sync
```
Comment (jeherve) [inline sync/class.jetpack-sync-defaults.php]: @gravityrail We'll need the `@since 5.7.0` line here. Do you think you could add it in a quick PR?

Thanks!

Comment (gravityrail) [reply sync/class.jetpack-sync-defaults.php]: Done: https://github.com/Automattic/jetpack/pull/8349

**Pull request merged**